### ISSUE TITLE
Stopped working entirely

### ISSUE THREAD

**Issue by Kylotan** [mozilla/contain-facebook#84]
It worked when I installed it initially, but today it doesn't.

No idea how to reproduce - all I know is that opening a new tab and navigating to Facebook doesn't open in a container any more, and that uninstalling/reinstalling the extension is no longer logging me out like it did the first time.

Currently running on Quantum 59.0.2.

The only other extension I run is Privacy Badger. There are also 4 deactivated legacy extensions.

**Comment by stoically** [comment]
Sorry it stopped working for you. Can you please provide the following information so we might be able to reproduce this?

- Facebook Container Version
- Operating System + Version
- Other installed Add-ons + Version: To Copy&Paste the full list of your Add-ons you can navigate to "about:support" and scroll down to "Extensions". This information would be espcially useful, so we can test with the exact same combination of installed / disabled extensions.

Thanks!

**Comment by Kylotan** [comment]
Facebook Container	1.2.5
Privacy Badger	2018.3.21

And disabled:
Pocket	3.0.6.1-signed
REST Easy	0.3.1.1-signed
SQLite Manager	0.8.3.1-signed.1-signed
Tab Groups	2.1.4

Windows 8.1, 64bit.

**Comment by stoically** [comment]
@Kylotan Thanks. Can you please provide the following additional informations:

- Is "Firefox will: Never remember history" in your Firefox Preferences/Options under "Privacy & Security History" selected?
- Are you using Firefox in a Private Window?

**Comment by Kylotan** [comment]
The answer to both is no. It's worth me repeating that the Facebook container worked fine when I first installed it. It's just stopped working since then, without me having made any changes to settings or add-ons.

**Comment by stoically** [comment]
Thanks again. That's really odd, I can't see why Facebook Container should just stop working given the informations you provided. Could you please open your Firefox Preferences/Options and under "Tabs" click on the "Settings" button right beside "Enable Container Tabs" (or alternatively navigate to `about:preferences#containers`) and comment here all Names of the Containers listed? A Screenshot would be fine too.

**Comment by Kylotan** [comment]
It lists 4 containers: Personal, Work, Banking, and Shopping. I've never used any of them (as I didn't know the feature existed until the Facebook one was announced).

**Comment by Kylotan** [comment]
(Also, I don't have a Settings button in the "Tabs" part of my options. Just 4 checkboxes.)

**Comment by stoically** [comment]
Oh. Now, that the detail about the missing settings button is interesting. This means that Firefox didn't activate the internal Container Feature correctly despite you installing an Extension that requires Containers.

You can manually activate the feature by navigating to `about:config`, and toggling the configurations `privacy.userContext.enabled` and `privacy.userContext.ui.enabled` to `true`. After that Facebook Container should work as expected.

Since this is a Firefox Bug it would be great if could file an Issue with all that information on http://bugzilla.mozilla.org/ too and leave the link here as reference, if you do so. Thanks!

**Comment by Kylotan** [comment]
I (re-)activated the feature, and it revealed the button in the settings, but unfortunately the Facebook Container add-on still doesn't work properly. I removed and re-installed it, but opening a tab and navigating to Facebook is still opening it normally, my cookies weren't cleared, and it doesn't open in a container.

**Comment by SoftVision-PaulOiegas** [comment]
> It lists 4 containers: Personal, Work, Banking, and Shopping. 

In order for Facebook Container to work, you need a "Facebook" container. It should be created at install so I don't know how you are missing it. Maybe it got deleted somehow?

**Comment by SoftVision-PaulOiegas** [comment]
Also, can you throw us an `about:support` page copy? We could check the prefs modified on your profile in our testing environment to check if any of them is generating this issue.

**Comment by stoically** [comment]
@Kylotan Unfortunately it seems that something messed up your Firefox profile regarding the Container feature. Could you provide the content of `privacy.userContext.extension` from `about:config` after installation of the Facebook Container Add-on? The next step would be to test with a new profile; you can easily create and start one from `about:profiles`

@SoftVision-PaulOiegas That's correct, the Facebook should've been created. But even if the container got deleted; after disabling/enabling, browser restart or Add-on reinstallation, the container should get created again. So it seems that for some reason the Container Feature can't be used anymore from the Facebook Container Add-in in @Kylotan's current profile.

**Comment by SoftVision-PaulOiegas** [comment]
We'll double check Privacy Badger in combination with Facebook Container. I don't see him mentioning about disabling it and I have a hunch. 😃

**Comment by Kylotan** [comment]
I just restarted the browser for an unrelated reason and now the Facebook container is back and working. I may have done it while Privacy Badger was disabled, but I'm not sure, sorry. But the container worked with Privacy Badger during the first install and is working fine with Privacy Badger enabled once more, so it would be odd if that were the culprit.

**Comment by stoically** [comment]
I've also tested with Privacy Badger and for me it works in combination with Facebook Container. Since it works now again - which is great - it'll probably be hard to find the original cause. Next time I should ask for a complete `about:support` copy earlier, like @SoftVision-PaulOiegas did. Oh well. Closing. :)

@SoftVision-PaulOiegas Feel free to reopen if you find a way to reproduce this

**Comment by Kylotan** [comment]
Not sure if same problem or different, so I'm adding here...

I went away from my computer for a few days, then came back and tried to navigate to Facebook via the URL bar autocomplete, - to http://facebook.com/ - and it showed the Facebook login page outside of a container. I was able to get 'in' to the container by right-clicking one of the links on that page and choosing 'Open link in new container tab > Facebook', so the container is still working - just not picking up the initial URLs properly.

All same config/environment as noted above.
